tools/cldr-code/src/main/java/org/unicode/cldr/util/UnitConverter.java

```diff
@@ -2236,4 +2253,30 @@ public String resolve(String unit) {
         String resolved = unitId.resolve().toString();
         return getStandardUnit(resolved.isBlank() ? unit : resolved);
     }
+
+    public String format(
+            final String languageTag,
+            Rational outputAmount,
+            final String unit,
+            UnlocalizedNumberFormatter nf3) {
+        final CLDRConfig config = CLDRConfig.getInstance();
+        Factory factory = config.getCldrFactory();
+        int pos = languageTag.indexOf("-u");
+        String localeBase =
+                (pos < 0 ? languageTag : languageTag.substring(0, pos)).replace('-', '_');
```

Comment:
better not to propagate locale string slicing

```suggestion
    final String localeBase = ULocale.forLanguageTag(languageTag).getBaseName();
```

something like that?

Reply:
I'm planning on another PR anyway, and will look at this. Since the beta is already tagged, would you mind approving now?